autopush_rs/src/client.rs

```diff
@@ -128,15 +153,26 @@ where
     /// the various state behind the server. This provides transitive access to
     /// various configuration options of the server as well as the ability to
     /// call back into Python.
-    pub fn new(ws: T, srv: &Rc<Server>) -> Client<T> {
+    pub fn new(ws: T, srv: &Rc<Server>, uarx: Receiver<Result<String>>) -> Client<T> {
         let srv = srv.clone();
         let timeout = Timeout::new(srv.opts.open_handshake_timeout.unwrap(), &srv.handle).unwrap();
+
+        // Pull out the user-agent, which we should have by now
+        let uastr = match uarx.wait() {
```

Comment:
In general calling `wait` can cause problems if it accidentally blocks, perhaps this could switch to `poll` which is nonblocking to assert that no blocking happens but it's still ready?